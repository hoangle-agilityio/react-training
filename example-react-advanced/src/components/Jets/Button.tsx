import { memo } from 'react';

interface ButtonProps {
  buttonName: string;
  className: string;
  onClick?: () => void;
}

const Button = ({ buttonName, className, onClick }: ButtonProps): JSX.Element => {
  return (
    <button data-testid="button" className={className} onClick={onClick}>
      <span className="button-title">{buttonName}</span>
    </button>
  );
};

export default memo(Button);
