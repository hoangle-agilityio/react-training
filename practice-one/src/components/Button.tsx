interface ButtonProps {
  typeButton: "button" | "submit" | "reset";
  size: "sm" | "md"| "lg";
  buttonColor: "light" | "info" | "danger";
  customize?: string;
  label: string;
}

export default function Button({ typeButton, size, buttonColor, customize, label }: ButtonProps) {
  const separateClass = customize ? customize : "";

  return (
    <button
      type={typeButton}
      className={`${separateClass} btn btn-${buttonColor} btn--${size}`}
    >
      {label}
    </button>
  );
}
