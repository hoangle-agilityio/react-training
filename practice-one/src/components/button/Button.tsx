import "./button.scss";

interface ButtonProps {
  typeButton: "button" | "submit" | "reset";
  size?: "sm" | "md"| "lg";
  buttonColor: "light" | "info" | "danger";
  customize?: string;
  label: string;
}

export default function Button({ typeButton, size, buttonColor, customize, label }: ButtonProps) {
  const separateClass: string = customize ? customize : "";
  const buttonSize: string = size ? `btn--${size}` : "";

  return (
    <button
      type={typeButton}
      className={`${separateClass} btn btn-${buttonColor} ${buttonSize}`}
    >
      {label}
    </button>
  );
}
