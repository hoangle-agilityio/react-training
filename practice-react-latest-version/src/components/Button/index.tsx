// Libs
import * as React from "react";
import { memo } from "react";
import { Button as ReactButton, ButtonProps } from "@chakra-ui/react";

interface Props {
  label: React.ReactNode;
  disabled?: boolean;
  type?: "primary" | "secondary" | "text";
  styles?: ButtonProps;
  onClick?: () => void;
}

const Button = ({
  label,
  disabled = false,
  type = "primary",
  styles,
  onClick,
}: Props) => {
  let props = null;
  switch (type) {
    case "primary":
      props = {
        bgColor: "blue.300",
        color: "white.100",
      };
      break;
    case "secondary":
      props = {
        bgColor: "transparent",
        color: "red.200",
      };
      break;
    default:
      props = {
        bgColor: "transparent",
      };
  }

  return (
    <ReactButton
      {...props}
      {...styles}
      fontSize="base"
      fontFamily="medium"
      fontWeight="500"
      _hover={{
        opacity: "70%",
        bgColor: `${type === "secondary" ? "gray.200" : null}`,
      }}
      isDisabled={disabled}
      onClick={onClick}
    >
      {label}
    </ReactButton>
  );
};

export default memo(Button);
