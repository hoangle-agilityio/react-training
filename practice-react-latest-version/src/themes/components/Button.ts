export const Button = {
  baseStyle: {
    fontFamily: "medium",
    fontWeight: "500",
    _hover: {
      opacity: "70%",
    },
  },
  variants: {
    primary: {
      bgColor: "background.primary",
      color: "brand.100",
      fontSize: "base",
    },
    secondary: {
      bgColor: "transparent",
      color: "text.reversal",
      fontSize: "base",
      _hover: {
        bgColor: "background.default",
      },
    },
  },
  defaultProps: {
    variant: "primary",
  },
};
