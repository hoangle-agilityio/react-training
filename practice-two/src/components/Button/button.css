.btn {
  border: none;
  border-radius: 5px;
  margin: 10px;
  cursor: pointer;
}

.button-edit {
  background-color: #037BFF;
  color: #FFFFFF;
  padding: 10px 20px;
  font-size: 14px;
}

.button-delete {
  background-color: #DC3545;
  color: #FFFFFF;
  padding: 10px 20px;
  font-size: 14px;
}
