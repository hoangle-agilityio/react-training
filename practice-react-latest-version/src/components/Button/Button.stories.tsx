// Libs
import * as React from "react";
import { ComponentMeta, ComponentStory } from "@storybook/react";

// Components
import Button from "components/Button";

export default {
  title: "Components/Button",
  component: Button,
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  ...Primary.args,
  label: "+ Add customer",
  type: "primary",
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: "Cancel",
  type: "secondary",
};

export const Text = Template.bind({});
Text.args = {
  label: "Edit",
  type: "text",
  styles: {
    color: "blue.200",
  },
};
