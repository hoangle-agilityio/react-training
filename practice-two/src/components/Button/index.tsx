import { memo } from "react";
import "./button.css";

interface ButtonProps {
  buttonName: string;
  type: string;
  onClick?: () => void;
}

function Button({ buttonName, type, onClick }: ButtonProps): JSX.Element {
  return (
    <button className={`btn btn-${type}`} onClick={onClick}>
      {buttonName}
    </button>
  );
}

export default memo(Button);
