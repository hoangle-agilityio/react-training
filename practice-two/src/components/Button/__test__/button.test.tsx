import Button from "../index";
import { cleanup, fireEvent, render } from "@testing-library/react";

const buttonProps = {
  buttonName: "Add User",
  type: "success",
}

describe('Button Component', () => {
  afterEach(cleanup);

  it('should be render button correctly', () => {
    const { getByRole } = render(<Button {...buttonProps} />);

    expect(getByRole('button').textContent).toBe('Add User');
  });

  it('should be click event', () => {
    const onClick = jest.fn();
    const { getByText } = render(<Button {...buttonProps} onClick={onClick} />);

    fireEvent.click(getByText('Add User'));
    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
