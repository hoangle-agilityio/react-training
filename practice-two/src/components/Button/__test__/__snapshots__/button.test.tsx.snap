// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should be click event 1`] = `
<button
  class="btn btn-success"
>
  Add User
</button>
`;

exports[`Button Component should be render button correctly 1`] = `
<button
  class="btn btn-success"
>
  Add User
</button>
`;
